lib/optimize/UglifyJsPlugin.js

```diff
@@ -31,57 +73,51 @@ class UglifyJsPlugin {
-				let files = [];
-				chunks.forEach((chunk) => files.push.apply(files, chunk.files));
-				files.push.apply(files, compilation.additionalChunkAssets);
-				files = files.filter(ModuleFilenameHelpers.matchObject.bind(undefined, options));
-				files.forEach((file) => {
-					let oldWarnFunction = uglify.AST_Node.warn_function;
-					let warnings = [];
-					let sourceMap;
+				var files = [];
+				for(var indexChunk = 0; indexChunk < chunks.length; indexChunk++) {
+					fastApply(files.push, files, chunks[indexChunk].files);
+				}
+				fastApply(files.push, files, compilation.additionalChunkAssets);
+				for(var indexFile = 0; indexFile < files.length; indexFile++) {
+					var file = files[indexFile];
+
+					if(!ModuleFilenameHelpers.matchObject(options, file)) {
+						continue;
+					}
+
+					var oldWarnFunction = uglify.AST_Node.warn_function;
+					var warnings = [];
+					var sourceMap;
 					try {
-						let asset = compilation.assets[file];
+						var asset = compilation.assets[file];
 						if(asset.__UglifyJsPlugin) {
 							compilation.assets[file] = asset.__UglifyJsPlugin;
 							return;
 						}
-						let input;
-						let inputSourceMap;
+						var input;
+						var inputSourceMap;
 						if(options.sourceMap) {
 							if(asset.sourceAndMap) {
-								let sourceAndMap = asset.sourceAndMap();
+								var sourceAndMap = asset.sourceAndMap();
 								inputSourceMap = sourceAndMap.map;
 								input = sourceAndMap.source;
 							} else {
 								inputSourceMap = asset.map();
 								input = asset.source();
 							}
 							sourceMap = new SourceMapConsumer(inputSourceMap);
-							uglify.AST_Node.warn_function = (warning) => { // eslint-disable-line camelcase
-								let match = /\[.+:([0-9]+),([0-9]+)\]/.exec(warning);
-								let line = +match[1];
-								let column = +match[2];
-								let original = sourceMap.originalPositionFor({
-									line: line,
-									column: column
-								});
-								if(!original || !original.source || original.source === file) return;
-								warnings.push(warning.replace(/\[.+:([0-9]+),([0-9]+)\]/, "") +
-									"[" + requestShortener.shorten(original.source) + ":" + original.line + "," + original.column + "]");
-							};
+							uglify.AST_Node.warn_function = warnFunctionSourceMap(sourceMap, file, warnings, requestShortener);
 						} else {
 							input = asset.source();
-							uglify.AST_Node.warn_function = (warning) => { // eslint-disable-line camelcase
-								warnings.push(warning);
-							};
+							uglify.AST_Node.warn_function = warnFunction(warnings);
 						}
 						uglify.base54.reset();
-						let ast = uglify.parse(input, {
+						var ast = uglify.parse(input, {
 							filename: file
 						});
 						if(options.compress !== false) {
 							ast.figure_out_scope();
-							let compress = uglify.Compressor(options.compress || {
+							var compress = uglify.Compressor(options.compress || {
```

Comment:
Shouldn't a decision be made about that first?  There are a dozen PR's currently up moving `var` to `let` and `const`.